src/targets/gpu/device/include/migraphx/gpu/device/visit.hpp

```diff
@@ -176,7 +176,12 @@ template <index_int N, class T, class... Ts>
 auto hip_vec_visit_all(T&& x, Ts&&... xs)
 {
     return [&](auto f) {
-        hip_visit_all_impl(get_shape(x),
+        auto sx   = get_shape(x);
+        auto lens = sx.lens();
+        assert(lens.back() % N == 0);
```

Comment:
Actually you need to check the stride is 1 as well.

Reply:
fixed